doc/CMakeLists.txt

```diff
@@ -93,7 +123,187 @@ FOREACH(_dot_prm ${_dot_prm_files})
+  # directory.
+  ADD_CUSTOM_COMMAND(
+    OUTPUT
+      ${CMAKE_CURRENT_BINARY_DIR}/${_image}
+    COMMAND
+      cp
+          ${CMAKE_SOURCE_DIR}/${_image}
+          ${CMAKE_CURRENT_BINARY_DIR}/${_image}
+    DEPENDS
+      ${CMAKE_SOURCE_DIR}/${_image}
+    COMMENT
+      "Copying ${_image} to build directory."
+    )
+
+  # Then we also need a target for the operation above. Target
+  # names may not contain slashes, so we need to escape those
+  STRING(REGEX REPLACE "[^a-zA-Z0-9]" "" _image_escaped "${_image}")
+  ADD_CUSTOM_TARGET(build_image_${_image_escaped}
+    DEPENDS
+      ${CMAKE_CURRENT_BINARY_DIR}/${_image}
+    )
+
+  # Finally add the target to the dependencies of the higher-level
+  # target
+  ADD_DEPENDENCIES(build_images build_image_${_image_escaped})
+ENDFOREACH()
+
+
+# Step 6: We have one image file that we need to create -- the plugin
+#         graph. This happens by calling the 'neato' program with
+#         the plugin graph input file generated by ASPECT.
+#
+#         Right now, this file is put into the repository, so we don't
+#         need to generate it. As a consequence, just add the rule
+#         whenever actually necessary. (If the file is found, then it's
+#         already copied into the build directory in step 5 above.)
+IF (NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/manual/plugin_graph.svg)
+  ADD_CUSTOM_COMMAND(
+    OUTPUT
+      ${CMAKE_CURRENT_BINARY_DIR}/doc/manual/plugin_graph.svg
+    COMMAND
+      neato
+        ${CMAKE_CURRENT_SOURCE_DIR}/manual/plugin_graph.dot
+        -Tsvg
+        -o ${CMAKE_CURRENT_BINARY_DIR}/doc/manual/plugin_graph.svg
+    DEPENDS
+      ${CMAKE_CURRENT_SOURCE_DIR}/manual/plugin_graph.dot
+    COMMENT
+      "Generating the plugin graph image."
+  )
+
+  ADD_CUSTOM_TARGET(build_plugin_graph_svg
+    DEPENDS
+      ${CMAKE_CURRENT_BINARY_DIR}/doc/manual/plugin_graph.svg
+  )
+
+  # Add the step above as a dependency -- somewhat randomly -- to the
+  # image generation target above. This way we don't have to do another
+  # IF(NOT EXISTS...) when adding dependencies to the larger targets below.
+  ADD_DEPENDENCIES(build_images
+                   build_plugin_graph_svg)
+ENDIF()
+
+
+# Step 7: Another preparatory step: Copy the manual.bib file into the
+#         directory where we will build the manual.
+ADD_CUSTOM_COMMAND(
+  OUTPUT
+    ${CMAKE_CURRENT_BINARY_DIR}/doc/manual/manual.bib
+  COMMAND
+    cp
+      ${CMAKE_CURRENT_SOURCE_DIR}/manual/manual.bib
+      ${CMAKE_CURRENT_BINARY_DIR}/doc/manual/manual.bib
+    DEPENDS
+      ${CMAKE_CURRENT_SOURCE_DIR}/manual/manual.bib
+    COMMENT
+      "Copying the bibliography database."
+  )
+
+
+
+# Step 8: Run pdflatex to generate the pdf version of the manual. This
+#         obviously depends on all of the previous targets.
+ADD_CUSTOM_COMMAND(
+    OUTPUT
+      ${CMAKE_CURRENT_BINARY_DIR}/doc/manual/manual.pdf
+    WORKING_DIRECTORY
+      doc/manual
+    COMMAND
+      rm -r svg-inkscape
```

Comment:
For reasons not entirely clear to me, when building by hand in the build directory, it occasionally trips up on some of the svg-converted-to-pdf files. Nuking the directory makes it work reliably, so I put that into the recipe here as well. I'm not sure it's strictly necessary, but it probably also doesn't hurt: We run latex four times, and the conversions from svg to pdf happen only on the first occasion. I don't observe a noticeable slowdown.